component/src/main/java/org/wso2/extension/siddhi/io/cdc/util/CDCSourceUtil.java

```diff
@@ -0,0 +1,237 @@
+import org.wso2.siddhi.query.api.exception.SiddhiAppValidationException;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This class contains Util methods for the cdc extension.
+ */
+public class CDCSourceUtil {
+    public static Map<String, Object> getConfigMap(String username, String password, String url, String tableName,
+                                                   String historyFileDirectory, String siddhiAppName,
+                                                   String siddhiStreamName, int serverID, String serverName,
+                                                   String connectorProperties, int cdcSourceHashCode)
+            throws WrongConfigurationException {
+
+        Map<String, Object> configMap = new HashMap<>();
+
+        String host;
+        int port;
+        String database;
+
+        //Add schema specific details to configMap
+
+        String[] splittedURL = url.split(":");
+        if (!splittedURL[0].equalsIgnoreCase("jdbc")) {
+            throw new WrongConfigurationException("Invalid JDBC url: " + url);
+        } else {
+            switch (splittedURL[1]) {
+                case "mysql": {
+
+                    //Extract url details
+                    String regex = "jdbc:mysql://(\\w*|[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}):" +
+                            "(\\d++)/(\\w*)";
+                    Pattern p = Pattern.compile(regex);
+                    Matcher matcher = p.matcher(url);
+                    if (matcher.find()) {
+                        host = matcher.group(1);
+                        port = Integer.parseInt(matcher.group(2));
+                        database = matcher.group(3);
+
+                    } else {
+                        throw new WrongConfigurationException("Invalid JDBC url: " + url);
+                    }
+
+                    //Add extracted url details to configMap.
+                    configMap.put(CDCSourceConstants.DATABASE_PORT, port);
+                    configMap.put(CDCSourceConstants.TABLE_WHITELIST, database + "." + tableName);
+                    configMap.put(CDCSourceConstants.DATABASE_HOSTNAME, host);
+
+                    //Add other MySQL specific details to configMap.
+                    configMap.put(CDCSourceConstants.CONNECTOR_CLASS, CDCSourceConstants.MYSQL_CONNECTOR_CLASS);
+
+                    break;
+                }
+                default: {
+                    throw new WrongConfigurationException("Unsupported schema. Expected schema: mysql, Found: "
+                            + splittedURL[1]);
+                }
+
+            }
+
+            //Add general config details to configMap
+            configMap.put(CDCSourceConstants.DATABASE_USER, username);
+            configMap.put(CDCSourceConstants.DATABASE_PASSWORD, password);
+
+            if (serverID == -1) {
```

Comment:
Move these numbers to a constant and reuse it.

Reply:
Fixed with commit: `Fix some suggested issues in PR#1.` 8a89f22